src/vehicles/RosVehicle.lua

```diff
@@ -177,10 +193,69 @@ function RosVehicle:getLaserFrameNode()
         -- and apply a transform to the self.laser_frame_1
         local tran_x, tran_y, tran_z = spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.x, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.y, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.z
         local rot_x, rot_y, rot_z = spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.x, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.y, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.z
-        local laser_frame_1 = frames.create_attached_node(instance_veh.cameraNode, self:getFullName(), tran_x, tran_y, tran_z, rot_x, rot_y, rot_z)
+        local laser_frame_1 = frames.create_attached_node(spec.instance_veh.cameraNode, self:getFullName(), tran_x, tran_y, tran_z, rot_x, rot_y, rot_z)
         spec.LaserFrameNode = laser_frame_1
         return spec.LaserFrameNode
     else
         return nil
     end
 end
+
+
+function RosVehicle:pubImu(ros_time, pub_imu)
+
+    local spec = self.spec_rosVehicle
+    -- retrieve the vehicle node we're interested in
+    local veh_node = self.components[1].node
```

Comment:
Should `veh_node` be checked for `nil` (ie: what if `self.components[1].node == nil`)?

Reply:
Yes, I've added an condition to check it.